pkg/volume/volume_linux.go

```diff
@@ -46,23 +47,29 @@ func SetVolumeOwnership(mounter Mounter, fsGroup *int64, fsGroupChangePolicy *v1
 
 	fsGroupPolicyEnabled := utilfeature.DefaultFeatureGate.Enabled(features.ConfigurableFSGroupPolicy)
 
-	klog.Warningf("Setting volume ownership for %s and fsGroup set. If the volume has a lot of files then setting volume ownership could be slow, see https://github.com/kubernetes/kubernetes/issues/69699", mounter.GetPath())
+	timer := time.AfterFunc(30*time.Second, func() {
+		klog.Warningf("Setting volume ownership for %s and fsGroup set. If the volume has a lot of files then setting volume ownership could be slow, see https://github.com/kubernetes/kubernetes/issues/69699", mounter.GetPath())
+	})
 
 	// This code exists for legacy purposes, so as old behaviour is entirely preserved when feature gate is disabled
 	// TODO: remove this when ConfigurableFSGroupPolicy turns GA.
 	if !fsGroupPolicyEnabled {
+		timer.Stop()
 		return legacyOwnershipChange(mounter, fsGroup)
 	}
 
 	if skipPermissionChange(mounter, fsGroup, fsGroupChangePolicy) {
 		klog.V(3).Infof("skipping permission and ownership change for volume %s", mounter.GetPath())
+		timer.Stop()
 		return nil
 	}
 
 	return walkDeep(mounter.GetPath(), func(path string, info os.FileInfo, err error) error {
 		if err != nil {
+			timer.Stop()
```

Comment:
I'm not sure you want to call timer.Stop() every time the func gets called. I think the intention here was to call it when the `walkDeep` object was done doing whatever it is that it does. It might be that you wanted to give `walkDeep` a reference too the timer so it can call stop when it's done.

Reply:
Yeah, this is wrong. Thanks for bring my attention for that. I just update the PR removing all the calls and adding a just single `defer` call.